File: WalkShopper/Mine/WSMineViewController.m
//
//  WSMineViewController.m
//  WalkShopper
//
//  Created by Ding Yi on 15/9/23.
//  Copyright (c) 2015年 Ding Yi. All rights reserved.
//

#import "WSMineViewController.h"
#import "WSMineHeaderView.h"
#import "WSUserInfoViewController.h"

@interface WSMineViewController () <UITableViewDataSource, UITableViewDelegate, WSMineHeaderViewDelegate>

@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property (weak, nonatomic) IBOutlet WSMineHeaderView *tableHeaderView;

@property (strong, nonatomic) NSMutableArray *dataSource;

@end

@implementation WSMineViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"我的";
    
    self.tableHeaderView.headerViewDelegate = self;
    self.tableView.tableFooterView = [UIView new];
    self.dataSource = [NSMutableArray arrayWithObjects:@[@"我的订单"], @[@"设置", @"关于"], nil];
}


#pragma mark UITableViewDataSource & UITableViewDelegate

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return self.dataSource.count;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [self.dataSource[section] count];
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
{
    return 20.0;
}

- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
{
    return CGFLOAT_MIN;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return 50.0f;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    cell.textLabel.text = self.dataSource[indexPath.section][indexPath.row];
    [cell setAccessoryType:UITableViewCellAccessoryDisclosureIndicator];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
}

#pragma mark - WSMineHeaderViewDelegate

- (void)showUserInfoViewController
{
    UIViewController *vc = [UIViewController ws_initViewControllerWithStoryBoard:@"Mine" withIdentifier:NSStringFromClass([WSUserInfoViewController class])];
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: WalkShopper/Mine/WSMineViewController.swift
//
//  WSMineViewController.swift
//  WalkShopper
//
//  Created by dulingkang on 16/2/3.
//  Copyright © 2016年 Ding Yi. All rights reserved.
//

import UIKit

class WSMineViewController: UIViewController, WSMineHeaderViewDelegate, UITableViewDelegate, UITableViewDataSource {

    @IBOutlet weak var headerView: WSMineHeaderView!
    @IBOutlet weak var tableView: UITableView!
    let kDefaultCellHeight: CGFloat = 44
    let kImageCellHeight: CGFloat = 72
    var rowHeight: CGFloat = 44
    var tableList: [(String, String)] {
        get{
            return  [("收货地址管理", ""),
                     ("客服", ""),
                     ("关于", "")
            ]
        }
    }

    
    override func viewDidLoad() {
        headerView.headerViewDelegate = self
    }
    
    //MARK: - header view delegate
    func showUserInfoViewController() {
        let userInfoVC = UIStoryboard(name: "Mine", bundle: nil).instantiateViewControllerWithIdentifier("WSUserInfoViewController")
        self.navigationController?.showViewController(userInfoVC, sender: nil)
    }
    
    //MARK: - tableview datasource
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 2
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        switch section {
        case 0:
            return 2
        default:
            return tableList.count
        }
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        var cell = MineCustomCell(style: .Detail, reuseIdentifier: MineCellType.Detail.identifier())
        if indexPath.section == 0 {
            switch indexPath.row {
            case 1:
                cell = MineCustomCell(style: .MultiButtons, reuseIdentifier: MineCellType.MultiButtons.identifier())
                rowHeight = kImageCellHeight
            default:
                cell.textLabel?.text = "我的订单"
                cell.detailTextLabel?.text = "查看全部订单"
                rowHeight = kDefaultCellHeight
                cell.accessoryType = .DisclosureIndicator
            }
        } else {
            cell.textLabel?.text = tableList[indexPath.row].0
            cell.detailTextLabel?.text = tableList[indexPath.row].1
            cell.accessoryType = .DisclosureIndicator
        }
        
        return cell
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        if indexPath.section == 0 && indexPath.row == 1 {
            return kImageCellHeight
        } else {
            return kDefaultCellHeight
        }
    }
}
